apps/els_dap/src/els_dap_breakpoints.erl

```diff
@@ -108,32 +108,38 @@ get_function_breaks(Module, Breaks) ->
 get_line_breaks(Module, Breaks) ->
     case Breaks of
         #{Module := #{line := Lines}} -> Lines;
-        _ -> []
+        _ -> #{}
     end.
 
 -spec do_line_breakpoints(
     node(),
     module(),
     #{line() => line_breaks()},
-    breakpoints()
+    breakpoints(),
+    boolean()
 ) ->
     breakpoints().
-do_line_breakpoints(Node, Module, LineBreakPoints, Breaks) ->
-    maps:map(
-        fun(Line, _) -> els_dap_rpc:break(Node, Module, Line) end,
-        LineBreakPoints
-    ),
+do_line_breakpoints(Node, Module, LineBreakPoints, Breaks, Set) ->
+    case Set of
+        true ->
+            maps:map(
+                fun(Line, _) -> els_dap_rpc:break(Node, Module, Line) end,
+                LineBreakPoints
+            );
+        false ->
+            ok
+    end,
     case Breaks of
         #{Module := ModBreaks} ->
             Breaks#{Module => ModBreaks#{line => LineBreakPoints}};
         _ ->
             Breaks#{Module => #{line => LineBreakPoints, function => []}}
     end.
 
--spec do_function_breaks(node(), module(), [function_break()], breakpoints()) ->
+-spec do_function_breaks(node(), module(), [function_break()], breakpoints(), boolean()) ->
     breakpoints().
-do_function_breaks(Node, Module, FBreaks, Breaks) ->
-    [els_dap_rpc:break_in(Node, Module, Func, Arity) || {Func, Arity} <- FBreaks],
+do_function_breaks(Node, Module, FBreaks, Breaks, Set) ->
+    [[els_dap_rpc:break_in(Node, Module, Func, Arity) || {Func, Arity} <- FBreaks] || Set],
```

Comment:
Can we please use a regular statement (or a function) instead of the double list comprehension? :)